File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q22-russian-doll-envelopes.js
var abc = function (envelopes) {
    envelopes.sort(function (a, b) {
        return a[0] - b[0];
    });
    var dp = [];
    var max = 0;
    for (var i = 0; i < envelopes.length; i++) {
        dp[i] = 1;
        for (var j = 0; j < i; j++) {
            if (envelopes[i][0] > envelopes[j][0] && envelopes[i][1] > envelopes[j][1]) {
                dp[i] = Math.max(dp[i], dp[j] + 1);
            }
        }
        max = Math.max(max, dp[i]);
    }
    return max;
}

File: ELEC881_Copilot/src/Function Name Masked/Python/q22-russian-doll-envelopes.py
class abc(object):
    def absd(self, envelopes):
        envelopes.sort(key=lambda x: (x[0], -x[1]))
        dp = []
        for w, h in envelopes:
            l, r = 0, len(dp)
            while l < r:
                m = (l + r) // 2
                if dp[m][1] < h:
                    l = m + 1
                else:
                    r = m
            if r == len(dp):
                dp.append([w, h])
            else:
                dp[r][1] = h
        return len(dp)